cmd/health_ready.go

```diff
@@ -0,0 +1,37 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ory/oathkeeper/sdk/go/oathkeeper/client/api"
+	"github.com/ory/x/cmdx"
+	"github.com/spf13/cobra"
+)
+
+// healthCmd represents the health command
+var healthReadyCmd = &cobra.Command{
+	Use:   "ready",
+	Short: "Checks if an ORY Oathkeeper deployment is ready",
+	Long: `Usage example:
+  oathkeeper health --endpoint=http://localhost:4456/ ready
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		client := newClient(cmd)
+
+		r, err := client.API.IsInstanceReady(api.NewIsInstanceReadyParams())
+		// If err, print err and exit 1
+		cmdx.Must(err, "%s", err)
+		// Print payload
+		fmt.Println(cmdx.FormatResponse(r.Payload))
+		// When healthy, ORY Oathkeeper always returns a status of "ok"
+		if r.Payload.Status != "ok" {
+			os.Exit(1)
```

Comment:
There should be an error message here. Something like:

```
fmt.Printf("The endpoint does not appear to be healthy, received HTTP Status Code %d and status %s\n", r.StatusCode, r.Payload.Status)
```

(I think the status code is `r.StatusCode`).

Reply:
The HTTP StatusCode isn't available. The internal API client returns the following struct
```
type IsInstanceReadyOK struct {
	Payload *models.SwaggerHealthStatus
}
```

If the HTTP status code isn't a 200, it will be printed as an error and the process will exit 1 which is handled by `cmdx.Must(err, "%s", err)`. Otherwise, the Payload is always printed so you can see the payload status

Reply:
Yeah, you're right!